native/src/boot/bootimg.hpp

```diff
@@ -646,14 +649,18 @@ struct boot_img {
     const uint8_t *extra;
     const uint8_t *recovery_dtbo;
     const uint8_t *dtb;
+    const uint8_t *vendor_ramdisk_table;
+    const uint8_t *bootconfig;
+
+    std::vector<std::tuple<std::unique_ptr<vendor_ramdisk_table_entry_v4>, format_t>> ramdisk_table_entries;
```

Comment:
after further consideration, I think this should be `std::vector<std::tuple<vendor_ramdisk_table_entry_v4*, format_t>>` to avoid some memcpy (if we can make sure the pointer will be always valid)

Reply:
done